server/src/main/java/org/opensearch/identity/IdentityService.java

```diff
@@ -48,4 +49,11 @@ public IdentityService(final Settings settings, final List<IdentityPlugin> ident
     public Subject getSubject() {
         return identityPlugin.getSubject();
     }
+
+    /**
+     * Gets a list of settings to register for extensions
+     */
+    public List<Setting<?>> getExtensionSettings() {
```

Comment:
@cwperks I have concerns about the general approach:
 - we bind plugins and extensions together: but those should evolve independently I think
 - the settings should be attached to particular extension: but the way those are applied now - to all extensions

But, realistically, in scope of the `security` plugin, we would probably need to know about the extensions, for better or worst. I would suggest to design this explicitly: introduce `ExtensionsAwarePlugin` interface that each plugin (including `security`) could implement and apply some actions (like add settings fe) on per-extension basis.

Reply:
Thank you for the feedback @reta! I don't currently see a straightforward way to make this extension specific, but I think that that's a good idea. The main problem is that these settings need to be defined before the `extensions.yml` file is read and at that point in time there is no `Extension`-like construct available to pass into a settings provider.

One thing that would be interesting to do would be something along the lines of: For all ActionExtensions, add a setting for REST Handler timeout. Even better, would be allowing different ActionExtensions to define different timeouts. 

Would it be reasonable to keep the approach now where these settings are applicable to all extensions, but extract this function from where it is in here (in `IdentityPlugin`) over to a new extension point (`ExtensionAwarePlugin`)?

Reply:
> With the example of REST Handler timeout as a setting, would that be applicable to all extensions or only to ActionExtensions that define REST handlers?

Tricky question, intuitive since this is "REST Handler timeout", it should apply to all  REST handlers (in case of extensions, "only to ActionExtensions that define REST handlers")